cli/groups.go

```diff
@@ -5,513 +5,582 @@ package cli
+func handleGroupRoleGet(cmd *cobra.Command, groupID string, args []string) {
+	if len(args) != 3 {
+		logUsageCmd(*cmd, usageGroupRolesGet)
+		return
 	}
-	for i := range cmdGroupsActions {
-		actionsCmd.AddCommand(&cmdGroupsActions[i])
+
+	roleID := args[0]
+	domainID := args[1]
+	token := args[2]
+
+	if roleID == all {
+		pageMetadata := smqsdk.PageMetadata{
+			Offset: Offset,
+			Limit:  Limit,
+		}
+		rs, err := sdk.GroupRoles(cmd.Context(), groupID, domainID, pageMetadata, token)
+		if err != nil {
+			logErrorCmd(*cmd, err)
+			return
+		}
+		logJSONCmd(*cmd, rs)
+		return
 	}
 
-	membersCmd := cobra.Command{
-		Use:   "members [add | list | delete]",
-		Short: "Members management",
-		Long:  "Members management: add, list, delete members",
+	r, err := sdk.GroupRole(cmd.Context(), groupID, roleID, domainID, token)
+	if err != nil {
+		logErrorCmd(*cmd, err)
+		return
 	}
-	for i := range cmdGroupMembers {
-		membersCmd.AddCommand(&cmdGroupMembers[i])
+	logJSONCmd(*cmd, r)
+}
+
+func handleGroupRoleUpdate(cmd *cobra.Command, groupID string, args []string) {
+	if len(args) != 4 {
+		logUsageCmd(*cmd, usageGroupRolesUpdate)
+		return
 	}
 
-	rolesCmd := cobra.Command{
-		Use:   "roles [create | get | update | delete | actions | members]",
-		Short: "Roles management",
-		Long:  "Roles management: create, update, retrieve roles and assign/unassign members to roles",
+	roleID := args[0]
+	newName := args[1]
+	domainID := args[2]
+	token := args[3]
+
+	r, err := sdk.UpdateGroupRole(cmd.Context(), groupID, roleID, newName, domainID, token)
+	if err != nil {
+		logErrorCmd(*cmd, err)
+		return
 	}
+	logJSONCmd(*cmd, r)
+}
 
-	rolesCmd.AddCommand(&actionsCmd)
-	rolesCmd.AddCommand(&membersCmd)
+func handleGroupRoleDelete(cmd *cobra.Command, groupID string, args []string) {
+	if len(args) != 3 {
+		logUsageCmd(*cmd, usageGroupRolesDelete)
+		return
+	}
 
-	for i := range cmdGroupsRoles {
-		rolesCmd.AddCommand(&cmdGroupsRoles[i])
+	roleID := args[0]
+	domainID := args[1]
+	token := args[2]
+
+	if err := sdk.DeleteGroupRole(cmd.Context(), groupID, roleID, domainID, token); err != nil {
+		logErrorCmd(*cmd, err)
+		return
 	}
+	logOKCmd(*cmd)
+}
 
-	cmd := cobra.Command{
-		Use:   "groups [create | get | update | delete | assign | unassign | users | channels ]",
-		Short: "Groups management",
-		Long:  `Groups management: create, update, delete group and assign and unassign member to groups"`,
+func handleGroupRoleActions(cmd *cobra.Command, groupID string, args []string) {
+	if len(args) < 1 {
+		logUsageCmd(*cmd, "cli groups <group_id> roles actions <operation> [args...]")
+		return
+	}
+
+	operation := args[0]
+	opArgs := args[1:]
+
+	switch operation {
+	case add:
+		handleGroupRoleActionsAdd(cmd, groupID, opArgs)
+	case list:
+		handleGroupRoleActionsList(cmd, groupID, opArgs)
+	case delete:
+		handleGroupRoleActionsDelete(cmd, groupID, opArgs)
+	case availableActions:
+		handleGroupRoleActionsAvailable(cmd, opArgs)
```

Comment:
list role actions does not require a groupID may be handle it elsewhere so that you are able to offer a proper help command. You can do this for the other entities as well.